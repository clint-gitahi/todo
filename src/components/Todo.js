import React, { Component } from 'react';

class Todo extends Component {
  constructor(props) {
    super(props);

    this.state = { isEditing: false, task: this.props.task };

    this.handleRemove = this.handleRemove.bind(this);
    this.toggleForm = this.toggleForm.bind(this);
    this.handleChange = this.handleChange.bind(this);
    // this.handleUpdate = this.handleUpdate.bind(this);
  }

  toggleForm() {
    this.setState({
      isEditing: !this.state.isEditing
    });
  }

  handleChange(event) {
    this.setState({
      [event.target.name]: event.target.value
    });
  }

  handleRemove() {
    this.props.removeTodo(this.props.id);
  }

  render() {
    console.log(this.state);
    let result;

    if (this.state.isEditing) {
      result = (
        <div>
          <form>
            <input
              type="text"
              name="task"
              value={this.state.task}
              onChange={this.handleChange}
            />
          </form>
        </div>
      );
    } else {
      result = (
        <div>
          <button onClick={this.toggleForm}>Edit</button>
          <button onClick={this.handleRemove}>X</button>
          <li>{this.props.task}</li>
        </div>
      );
    }
    return result;
  }
}

export default Todo;
