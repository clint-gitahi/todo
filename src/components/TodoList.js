import React, { Component } from 'react';
import Todo from './Todo';
import NewTodoForm from './newTodoForm';

class TodoList extends Component {
  constructor(props) {
    super(props);

    this.state = {
      todos: [
        { task: 'start leetcode' },
        { task: 'complete divide and conquer section' }
      ]
    };
  }

  render() {
    const todos = this.state.todos.map(todo => {
      return <Todo task={todo.task} />;
    });
    return (
      <div>
        <h1>
          Todo list <span>Simple Todo List App</span>
          <NewTodoForm />
          <ul>{todos}</ul>
        </h1>
      </div>
    );
  }
}

export default TodoList;
