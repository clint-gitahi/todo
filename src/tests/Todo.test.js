import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import Todo from '../components/Todo';
import TodoList from '../components/TodoList';

test('renders TODO without crashing', () => {
  const div = document.createElement('div');

  ReactDOM.render(<Todo />, div);
});

// using shallow.
test('render TODOLIST  without crashing', () => {
  shallow(<TodoList />);
});

it('renders todolist', () => {
  const wrapper = shallow(<TodoList />);
  const header = (
    <h1>
      Todo list <span>Simple Todo List App</span>
    </h1>
  );

  expect(wrapper.contains(header)).toEqual(true);
});
