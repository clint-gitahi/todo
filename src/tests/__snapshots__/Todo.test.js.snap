// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `ShallowWrapper {}`;

exports[`should test Todo component 1`] = `ShallowWrapper {}`;
